### PR TITLE
Fix inline images migration (Document_Item)

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Inline image migration to glpi_documents_items was not working as expected, only the first image of each item was migrated.
This was cause by the use of preg_match (stop at first match) instead of preg_match_all.

These changes fix the migration process and move it to 9.5.2.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: install/update_951_952.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * Update from 9.4.6 to 9.4.7
+ *
+ * @return bool for success (will die for most error)
+ **/
+function update951to952() {
```
Comment (trasher) [inline install/update_951_952.php:38]: wrong function name

File: install/update_951_952.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * Update from 9.4.6 to 9.4.7
```
Comment (trasher) [inline install/update_951_952.php:34]:
```suggestion
 * Update from 9.5.1 to 9.5.2
```

File: install/update_94_95.php [hunk 3]
```diff
@@ -1513,60 +1513,6 @@ function update94to95() {
...
-   /** A doc_item to rule them all! */
-   $migration->dropKey('glpi_documents_items', 'unicity');
-   $migration->migrationOneTable('glpi_documents_items');
-   $migration->addKey(
-      'glpi_documents_items',
-      ['documents_id', 'itemtype', 'items_id', 'timeline_position'],
-      'unicity',
-      'UNIQUE'
-   );
-   $migration->migrationOneTable('glpi_documents_items');
```
Comment (cedric-anne) [inline install/update_94_95.php:1525]: This should stay here. DB already migrated on GLPI 9.5.x have correct `unicity` key, no need to rebuilt it for them.

File: inc/update.class.php [hunk 4]
```diff
@@ -473,6 +473,10 @@ public function doUpdates($current_version = null) {
          case "9.5.0-dev":
             include_once "{$updir}update_94_95.php";
             update94to95();
+
+         case "9.5.1":
```
Comment (cedric-anne) [inline inc/update.class.php:477]:
```suggestion
         case "9.5.0":
         case "9.5.1":
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**